src/v/kafka/server/snc_quota_manager.cc

```diff
@@ -481,6 +575,20 @@ void snc_quota_manager::refill_buckets(const clock::time_point now) noexcept {
     _shard_quota.out.refill(now);
 }
 
+inoutpair<quota_t> snc_quota_manager::get_deficiency() const noexcept {
+    const auto f = [](
```

Comment:
can be marked `static const`

Reply:
what would marking `f` as `static` achieve? it doesn't seem to have any state, for instance.

Reply:
I guess nothing other then avoiding the initialization of `f` once per invocation. Logically I see it as a non-changing constant for the duration of the program, I generally mark things like that as `static + const`. Effectively, its the same as just making this lambda a function.

Reply:
> I guess nothing other then avoiding the initialization of f once per invocation
i think we are probably underestimating the compiler if we believe that it isn't inline this.